charts/redpanda/values.yaml

```diff
@@ -640,6 +640,19 @@ statefulset:
   # -- Additional flags to pass to redpanda,
   additionalRedpandaCmdFlags: []
 #    - --unsafe-bypass-fsync
+  # -- Termination grace period in seconds is time required to execute preStop hook
+  # which puts particular Redpanda Pod (process/container) into maintenance mode.
+  # Before settle down on particular value please put Redpanda under load and perform
+  # rolling upgrade or rolling restart. That value needs to accommodate two processes:
+  # * preStop hook needs to put Redpanda into maintenance mode
+  # * after preStop hook Redpanda needs to handle gracefully SIGTERM signal
+  #
+  # Both processes are executed sequentially where preStop hook has hard deadline in the
+  # middle of terminationGracePeriodSeconds.
```

Comment:
Does this mean that the `preStop` hook only has 45 seconds by default and the rest is allocated to the `SIGTERM`?

Reply:
Great question. It was working like that, but latest changes https://github.com/redpanda-data/helm-charts/commit/2b87951931b00d5c19cf062f7113238860c99b1e#diff-defacebd1c27c71d9807d81685dd0838ddd8392ca6084491962f85dca6bb2307L68 changed that behavior.

@joejulian Why we don't cap the preStop hook?

Reply:
We where using timeout command 

REF: 
https://man7.org/linux/man-pages/man1/timeout.1.html

Reply:
@joejulian send me the exact place where we are using timeout function:
* https://github.com/redpanda-data/helm-charts/blob/4f599dcc17d941b3e83f84c292faf51f0740e122/charts/redpanda/templates/statefulset.yaml#L177
* https://github.com/redpanda-data/helm-charts/blob/4f599dcc17d941b3e83f84c292faf51f0740e122/charts/redpanda/templates/statefulset.yaml#L185

You can find that `{{ div .Values.statefulset.terminationGracePeriodSeconds 2 }}` is used to divide period by 2.

Reply:
So yes, if the `terminationGracePeriodSeconds` is configured to the default of 90 seconds, preStop will always be stopped by 45 seconds after which redpanda will get its SIGTERM.

If preStop finishes earlier, redpanda will get whatever's left at that time.